### PR TITLE
Support multiple instance by adding instance id

### PR DESCRIPTION
**Pull request by jia-guo**
- add instanceId as the identifier of the instance
- encode instanceId in the state params of the `authUrl`
- check instanceId when `messageHandler` is executed

### CODE REVIEW

File: test/unit/sdk.test.js [hunk 1]
```diff
@@ -140,7 +144,7 @@ describe('sdk', () => {
           isSmartcarHosted: false,
           code: 'super-secret-code',
           error: undefined,
-          state: 'some-state',
+          state: window.btoa(`{"instanceId":"${smartcar.instanceId}","originalState":"some-state"}`),
```
Comment (sankethkatta) [inline test/unit/sdk.test.js]: Can we use the `getEncodedState()` function for these? If we need to customize the state we can make the helper have a default value:
```js
const getEncodedState = (instanceId, originalState = originalState) => ...
```

**Comment by sankethkatta** [comment]
The rest of the tests look good to me! Just the couple comments above. And we also need to put the right commit message to push up the new version.

**Comment by jia-guo** [comment]
> The rest of the tests look good to me! Just the couple comments above. And we also need to put the right commit message to push up the new version.

Got it! Thanks so much for reviewing the PR!

File: package-lock.json [hunk 2]
```diff
@@ -1010,6 +1010,14 @@
         "url": "0.10.3",
         "uuid": "3.1.0",
         "xml2js": "0.4.19"
+      },
+      "dependencies": {
+        "uuid": {
```
[the remark this answers is not in the document]
Comment (jia-guo) [reply package-lock.json:1015]: Just realized that this uuid is a dependency of a dependency :joy:. I've removed package-lock and node_modules and re-run npm install, the current one should be alright.

File: test/unit/sdk.test.js (hunk 1 above)
Comment (jia-guo) [reply test/unit/sdk.test.js]: Good idea! Much cleaner.

**Comment by jia-guo** [comment]
@sankethkatta I tried out in mock ui & found it works as intended.

image

**Pull request merged**